src/components/calcite-button/calcite-button.tsx

```diff
@@ -169,20 +170,33 @@ export class CalciteButton {
 
   //--------------------------------------------------------------------------
   //
-  //  Private State/Props
+  //  Private Methods
   //
   //--------------------------------------------------------------------------
 
-  /** @internal */
-  /** if button type is present, assign as prop */
-  private type?: string;
+  @Method()
+  async focusCalciteButton() {
```

Comment:
Can we call the method `setFocus()`? Then we can just have a utility like this to handle focusing any of our components.

```ts
export function setElementFocus(el: HTMLCalciteActionElement | HTMLElement): void {
  if (!el) {
    return;
  }

  "setFocus" in el && typeof el.setFocus === "function" ? el.setFocus() : el.focus();
}```

Reply:
Is it expected to need to define each component that will use the exported function? I found that unless I did so I got a `Property 'setFocus' does not exist on type 'never'` even if the function was never imported into a component.